tools/fmt_diff/dune

```diff
@@ -0,0 +1,5 @@
+(executable
+  (name        fmt_diff)
+  (public_name fmt_diff)
```

Comment:
My impression is that `fmt-diff` would be more idiomatic than `fmt_diff`.

Also, is it self-inconsistent to have `ocamlformat` and `fmt-diff` executables, rather than `ocamlformat-diff`?

Reply:
It doesn't seem to be possible to name it `ocamlformat-diff`, the name of the executable has to be the same as one of the module and it is an invalid module name (it's only a warning but still), we could still copy it manually in the `.opam/<VERSION>/bin` but it is not as straightforward as using `dune install` and it overcomplicates the Makefile.

Reply:
Those exact lines give me:
```
dune build _build/dev/tools/ocamlformat-diff/ocamlformat-diff.exe _build/release/tools/ocamlformat-diff/ocamlformat-diff.exe
Don't know how to build _build/dev/tools/ocamlformat-diff/ocamlformat-diff.exe
Hint: did you mean _build/dev/tools/ocamlformat-diff/ocamlformat_diff.exe?
Makefile:48: recipe for target 'ocamlformat-diff' failed
make: *** [ocamlformat-diff] Error 1
```
I use `dune` 1.2.1 but I didn't see interesting changes in the 1.3 changelog, am I missing something else?

Reply:
That sounds like a bug. It used to work as suggested by @jberdine and @avsm

Reply:
Weirdly enough, `dune build _build/install/default/bin/ocamlformat-diff`seems to be working for me

Reply:
I get the same output as I did earlier (wihtout the .exe). What version of `dune` are you all using? If it is indeed fixed in dune-1.3 I don't know if it is worth it to upgrade the dependencies for `dune` and `opam` only for the tool name, as the main use for this tool is to be called by `git diff`.